CometServer/Modules/10-25/ApiBase.cs

```diff
@@ -172,7 +182,66 @@ protected ApiBase(IAppConfigService appConfigService, ICometHasStartedService co
+        /// <summary>
+        /// Authorizes a <paramref name="request"/> and calls the
+        /// <see cref="ICredentialsService.ResolveCredentials"/> to resolve and set the
+        /// <see cref="ICredentialsService.Credentials"/> to be used in the following pipeline
+        /// </summary>
+        /// <param name="credentialsService">
+        /// The <see cref="ICredentialsService"/> used to provide authorization and <see cref="Credentials"/>
+        /// services while handling a request
+        /// </param>
+        /// <param name="appConfigService">
+        /// The <see cref="IAppConfigService"/> used to read applicaton settings
+        /// </param>
+        /// <param name="request">
+        /// The <see cref="HttpRequest"/> that provides that should be ahtorized
+        /// </param>
+        /// <returns>
+        /// an awaitable <see cref="Task"/> that contains the value of the identifier uses to check authorization
+        /// </returns>
+        protected async Task<string> Authorize(IAppConfigService appConfigService, ICredentialsService credentialsService, HttpRequest request)
+        {
+            var userName = request.HttpContext.User.Identity!.Name;
+
+            if (!request.DoesAuthorizationHeaderMatches(JwtBearerDefaults.ExternalAuthenticationScheme))
+            {
+                await this.Authorize(appConfigService, credentialsService, userName);
+                return userName;
+            }
+
+            var claim = request.HttpContext.User.Claims
+                .FirstOrDefault(x => x.Type == appConfigService.AppConfig.AuthenticationConfig.ExternalJwtAuthenticationConfig.IdentifierClaimName);
+
+            if (claim != null)
+            {
+                switch ( appConfigService.AppConfig.AuthenticationConfig.ExternalJwtAuthenticationConfig.PersonIdentifierPropertyKind)
+                {
+                    case PersonIdentifierPropertyKind.ShortName:
+                        await this.Authorize(appConfigService, credentialsService, claim.Value);
+                        break;
+                    case PersonIdentifierPropertyKind.Iid:
+                        if (!Guid.TryParse(claim.Value, out var userId))
+                        {
+                            throw new AuthenticationException();
+                        }
+                        
+                        await this.Authorize(appConfigService, credentialsService, userId);
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(PersonIdentifierPropertyKind), "Unsupported PersonIdentifierPropertyKind");
+                }
+                
+                return claim.Value;
+            }
+
+            this.logger.LogWarning("Identifier claim {ClaimName} missing User Claims",
+                appConfigService.AppConfig.AuthenticationConfig.ExternalJwtAuthenticationConfig.IdentifierClaimName);
+
+            throw new AuthorizationException();
```

Comment:
should we not add textual description to exceptino, similar to log statemnt?

Reply:
Fixed